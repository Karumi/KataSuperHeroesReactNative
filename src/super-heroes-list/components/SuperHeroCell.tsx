import * as React from "react";
import { Image, StyleSheet, TouchableHighlight, View } from "react-native";
import { Text } from "react-native";
import LinearGradient from "react-native-linear-gradient";
import { SuperHero } from "../../core/model";

interface Props {
    readonly superHero: SuperHero;
    readonly onTap: (superHero: SuperHero) => void;
}

class SuperHeroCell extends React.Component<Props> {

    public render() {
        const superHero = this.props.superHero;
        return (
            <TouchableHighlight onPress={() => this.props.onTap(this.props.superHero)}>
                <View
                    style={styles.container}>
                    <Image
                        style={{
                            position: "absolute",
                            height: "100%",
                            width: "100%",
                            zIndex: 1,
                            resizeMode: "cover",
                        }}
                        source={{ uri: superHero.picture }}
                    />
                    <Text style={styles.name}>{this.props.superHero.name}</Text>
                    <LinearGradient style={styles.gradient} colors={["transparent", "#000"]} />
                </View>
            </TouchableHighlight>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        height: 200,
        justifyContent: "flex-end",
        zIndex: 0,
    },
    gradient: {
        height: 100,
        width: "100%",
        position: "absolute",
        zIndex: 2,
    },
    name: {
        fontSize: 20,
        margin: 16,
        color: "white",
        zIndex: 3,
    },
});
export default SuperHeroCell;
