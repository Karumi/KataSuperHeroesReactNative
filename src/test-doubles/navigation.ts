
export const mockPlainNavigation = {
    navigate: jest.fn(),
};

export const navigationWithMockedParams = {
    navigate: jest.fn(),
    getParam: jest.fn(),
};
