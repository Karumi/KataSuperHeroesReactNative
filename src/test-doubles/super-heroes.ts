import { SuperHero } from "../core/model";

export const anySuperHeroId: string = "Gambito";

export const anyListOfSuperHeroes: SuperHero[] = [
    new SuperHero("Scarlet Witch",
        "https://i.annihil.us/u/prod/marvel/i/mg/9/b0/537bc2375dfb9.jpg",
        false,
        "Scarlet Witch was born at the Wundagore base of the High Evolutionary, she and her twin "
        + "brother Pietro were the children of Romani couple Django and Marya Maximoff. The "
        + "High Evolutionary supposedly abducted the twins when they were babies and "
        + "experimented on them, once he was disgusted with the results, he returned them to"
        + " Wundagore, disguised as regular mutants."),
    new SuperHero("Iron Man",
        "https://i.annihil.us/u/prod/marvel/i/mg/c/60/55b6a28ef24fa.jpg",
        true,
        "Wounded, captured and forced to build a weapon by his enemies, billionaire "
        + "industrialist Tony Stark instead created an advanced suit of armor to save his "
        + "life and escape captivity. Now with a new outlook on life, Tony uses his money "
        + "and intelligence to make the world a safer, better place as Iron Man."),
    new SuperHero("Wolverine",
        "https://i.annihil.us/u/prod/marvel/i/mg/9/00/537bcb1133fd7.jpg",
        false,
        "Born with super-human senses and the power to heal from almost any wound, " +
        "Wolverine "
        + "was captured by a secret Canadian organization and given an unbreakable "
        + "skeleton and claws. Treated like an animal, it took years for him to control"
        + " himself. Now, he's a premiere member of both the X-Men and the Avengers."),
];

export const anyRegularSuperHero = anyListOfSuperHeroes[0];

export const anyAvenger = anyListOfSuperHeroes[1];
