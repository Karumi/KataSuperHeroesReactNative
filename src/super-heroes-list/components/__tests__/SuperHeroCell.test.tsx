import * as React from "react";
import * as renderer from "react-test-renderer";

import { anyAvenger, anyRegularSuperHero } from "../../../test-doubles/super-heroes";
import SuperHeroCell from "../SuperHeroCell";

it("renders a regular super hero and does not show the avengers badge", () => {
    const component = renderer
        .create(<SuperHeroCell onTap={jest.fn()} superHero={anyRegularSuperHero} />)
        .toJSON();
    expect(component).toMatchSnapshot();
});

it("renders an avenger super hero and shows the avengers badge", () => {
    const component = renderer
        .create(<SuperHeroCell onTap={jest.fn()} superHero={anyAvenger} />)
        .toJSON();
    expect(component).toMatchSnapshot();
});
