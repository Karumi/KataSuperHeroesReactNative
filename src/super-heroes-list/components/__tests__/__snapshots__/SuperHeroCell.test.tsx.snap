// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a regular super hero and does not show the avengers badge 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "height": 200,
        "justifyContent": "flex-end",
        "zIndex": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.annihil.us/u/prod/marvel/i/mg/9/b0/537bc2375dfb9.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "resizeMode": "cover",
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "margin": 16,
          "zIndex": 3,
        }
      }
    >
      Scarlet Witch
    </Text>
    <ExponentLinearGradient
      colors={
        Array [
          0,
          4278190080,
        ]
      }
      endPoint={undefined}
      locations={null}
      startPoint={undefined}
      style={
        Object {
          "height": 100,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        }
      }
    />
  </View>
</View>
`;

exports[`renders an avenger super hero and shows the avengers badge 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "height": 200,
        "justifyContent": "flex-end",
        "zIndex": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.annihil.us/u/prod/marvel/i/mg/c/60/55b6a28ef24fa.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "resizeMode": "cover",
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "margin": 16,
          "zIndex": 3,
        }
      }
    >
      Iron Man
    </Text>
    <ExponentLinearGradient
      colors={
        Array [
          0,
          4278190080,
        ]
      }
      endPoint={undefined}
      locations={null}
      startPoint={undefined}
      style={
        Object {
          "height": 100,
          "position": "absolute",
          "width": "100%",
          "zIndex": 2,
        }
      }
    />
    <Image
      source="avengersBadge"
      style={
        Object {
          "alignSelf": "flex-end",
          "bottom": 16,
          "height": 101,
          "position": "absolute",
          "right": 16,
          "width": 90,
          "zIndex": 3,
        }
      }
    />
  </View>
</View>
`;
